### PR TITLE
fix: add missing README information and clean up types

### PR DESCRIPTION
**Pull request by Parkreiner**
Little bit of cleanup to make the repo easier to maintain.

## Changes made
- Updated the main contributing guide to include release steps
- Cleaned up TypeScript implementation
   - Turned strict mode on
   - Removed all `any` types
   - Updated all runtime code to account for type changes

## Notes
- Going to look into adding images or even a gif for the release steps; will look into this on Monday
- I wanted to get the `tsconfig.json` file cleaned up, too (the compiler settings are technically invalid, even though nothing seems to break), but that took me down a rabbi thole of ESM-CSJ interop. Aiming to address that in a separate PR

### CODE REVIEW

**Comment by stirby** [comment]
@matifali @Parkreiner can we cut a release as soon as this is merged?

**Comment by Parkreiner** [comment]
> @matifali @Parkreiner can we cut a release as soon as this is merged?

Atif and I just cut a release this afternoon. Were there any major changes after that? I would think this PR is just infrastructure stuff that won't affect end-users

**Review by bcpeinhardt — APPROVED** [review]
This is super useful on boarding info 😎

File: README.md [hunk 1]
```diff
@@ -3,14 +3,14 @@
   Modules
   </h1>
 
-[Registry](https://registry.coder.com) | [Coder Docs](https://coder.com/docs) | [Why Coder](https://coder.com/why) | [Coder Enterprise](https://coder.com/docs/v2/latest/enterprise)
+[Module Registry](https://registry.coder.com) | [Coder Docs](https://coder.com/docs) | [Why Coder](https://coder.com/why) | [Coder Enterprise](https://coder.com/docs/v2/latest/enterprise)
```
Comment (matifali) [inline README.md:6]: Registry also hosts templates source code at https://registry.coder.com/templates

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -27,4 +50,25 @@ module "example" {
...
+2. Click "Draft a new release"
+3. Click the "Choose a tag" button and type a new release number in the format `v<major>.<minor>.<patch>` (e.g., `v1.18.0`). Then click "Create new tag".
+4. Click the "Generate release notes" button, and clean up the resulting README. Be sure to clean up any notes that would not be relevant to end-users (e.g., bumping dependencies).
+5. Once everything looks good, click the "Publish release" button.
+
+Once the release has been cut, a script will run to check whether there are any modules that will require that the new release number be published to Terraform. If there are any, a new pull request will automatically be generated. Be sure to approve this PR and merge it into the `main` branch.
+
+Following that, our automated processes will handle publishing new data for [`registry.coder.com`](https://github.com/coder/registry.coder.com/):
+
+1. Publishing new versions to Coder's [Terraform Registry](https://registry.terraform.io/providers/coder/coder/latest)
```
Comment (matifali) [inline CONTRIBUTING.md:70]: We do not publish anything to terraform registry.

**Review by matifali — APPROVED** [review]
Looking great for now. It may change once we implement independent versioning for each module. 
#157

**Review by matifali — CHANGES_REQUESTED** [review]
We do not publish modules to terraform registry. i.e registry.terraform.io

**Review by matifali — APPROVED** [review]
Thank you. :)

**Pull request merged**